### ISSUE TITLE
Question about minimal code for SKD

### ISSUE THREAD

**Issue by DeepLearningHB** [forjiuzhou/Spherical-Knowledge-Distillation#2]
Hi there, I've read your interesting paper. and thank you for sharing the codes.
I'm rewriting code to adequate RepDistiller format. and I have a question.

As mentioned in the paper, a Teacher's norm(l_avg) is needed to apply your idea. 
then we can make our new logit (f^_i(x) * l_avg, f^_j(x) * l_avg) 
but your simplified code doesn't seem to reflect these operations.

Is it replacing this operation by multiplying a constant between 2 and 3? 

Please understand that I did not fully understand the paper.  :)
thanks.

**Comment by forjiuzhou** [comment]
Thanks for your feedback. 

'Is it replacing this operation by multiplying a constant between 2 and 3?'
Yes. In our previous experiments, we find that student performance is robust with l_avg, as long as l_avg is larger than 2. And in most cases, the average of teacher logits norm is between 2 and 3. 

Since SKD require that the cross entropy loss with ground truth is also computed with normalized logits, the implementation would need some adjustment to comply with RepDistiller format. RepDistiller has three losses, CE, KD, and the 3rd distillation loss. I would recommend to incorporate CE and SKD loss in to one class. The loss weight would be 0,0,1, and the exact weight with CE and SKD would need to be hard encoded in the code.

**Comment by DeepLearningHB** [comment]
Thank you for your constructive reply. It's really helpful to fully understand.
